src/main/java/com/google/edith/servlets/Item.java

```diff
@@ -1,19 +1,40 @@
 package com.google.edith.servlets;
 
+import com.google.auto.value.AutoValue;
+
  /** Used to hold the properties of an item Entity in datastore. */
-public final class Item {
-  // TODO (malachibre): Add name, user, category, and receiptId fields.
-  // TODO (malachibre): Create a builder for this class inclduing checks 
-  //                    for valid assignments.
-  String name;
-  double price; 
-  long quantity;  
-  String date; 
- 
-  public Item(String name, double price, long quantity, String date) {
-    this.name = name;
-    this.price = price;
-    this.quantity = quantity;
-    this.date = date;
+@AutoValue
+public abstract class Item {
+  public abstract String name();
+  public abstract String userId();
+  public abstract String category();
+  public abstract double price(); 
+  public abstract long quantity();  
+  public abstract String date(); 
+  public abstract String receiptId();
```

Comment:
do we need a receiptId ?
if we have a receipt key then, receiptId might not be needed.

Reply:
Oh okay, I removed it.